core/unit_test/CMakeLists.txt

```diff
@@ -831,6 +825,18 @@ KOKKOS_ADD_EXECUTABLE_AND_TEST(
   SOURCES ${DEFAULT_DEVICE_SOURCES}
 )
 
+# FIXME_OPENMPTARGET and FIXME_OPENACC do not provide a MemorySpace that can be accessed from all ExecSpaces
+# FIXME_SYCL clock_tic does not give the correct timings for cloc_tic
+if (NOT (KOKKOS_ENABLE_OPENACC OR KOKKOS_ENABLE_OPENMPTARGET OR KOKKOS_ENABLE_SYCL))
+  KOKKOS_ADD_EXECUTABLE_AND_TEST(
+    UnitTest_SharedSpace
+    SOURCES
+      UnitTestMainInit.cpp
+      TestSharedSpace.cpp
+  )
+  set_property(TEST KokkosCore_UnitTest_SharedSpace PROPERTY RUN_SERIAL TRUE)
```

Comment:
@JBludau I ran into this configuration error with Trilinos when testing this PR:

```
CMake Error at kokkos/core/unit_test/CMakeLists.txt:837 (set_property):
  set_property given TEST names that do not exist:

    KokkosCore_UnitTest_SharedSpace
```

Reply:
Hm, hacked in a change to append the suffix for a quick check that would resolve the CMake error but no luck

```
CMake Error at kokkos/core/unit_test/CMakeLists.txt:837 (set_property):
  set_property given TEST names that do not exist:

    KokkosCore_UnitTest_SharedSpace.exe
```

Reply:
I think the .exe is not part of the target name but just appended to the executable
Anyway ... looks like we have to go through the hassle of adding/extending the utility to add a test that will be run in serial

Reply:
@JBludau alternatively we can request the Trilinos AT Cuda builds set the RUN_SERIAL option as part of the configuration (`-D KokkosCore_UnitTest_Default_MPI_1_SET_RUN_SERIAL=ON` is recognized on the configure line with Trilinos)
I thought adding the option directly here would just work and didn't realize extending the build system would be necessary. Let me know if you'd rather we just request this on the Trilinos testing side

Reply:
To be honest, I struggle with this decision, as I have no means of finding out if there will be issues where our tests in projects downstream of us need to be serialized. If we do it, then we do not have to worry ... sort of, we still do not know in which conditions the test is run (other stuff running on the same machine). If we fix it in Trilinos (via the configure time option you described), other projects might run into the same problem. I think the question is what we think is the better option in the long term, maybe @crtrott and @dalg24 have a better overview of the situation here.

Given the current situation, I would argue, that we do not give any guarantee that the kokkos test can be run in parallel and Trilinos should just test in serial. But I can understand, that in a big project you do not want to have this constraint ...

Reply:
I kind of expect that Trilinos is somewhat distinct among downstream projects incorporating Kokkos in running its tests. Others can probably pretty much take "Kokkos will pass its own tests" for granted, given that they're likely not pushing the boundaries of platforms, compilers, etc, the way Trilinos's own testing environment does.

My general take is that it would be good for Kokkos to declare the constraints of its own tests internally, but if that comes at a big expense relative to just fixing them up for our CI pipelines and those run by Trilinos, it may not be worth it.

Reply:
Look at
https://github.com/kokkos/kokkos/blob/aa98af236f69e90535f5db7572dbbac9d2c437ef/core/perf_test/CMakeLists.txt#L89